algorithms/CPlusPlus/Graphs/topological-sort.cpp

```diff
@@ -0,0 +1,140 @@
+//Topological sort to check order using dfs
+
+#include<bits/stdc++.h>
+
+using namespace std;
+
+typedef struct node {
+
+  int dest;
+
+}
+node;
+/*
+Using graph class with vector of nodes to represent DAG(Directed acyclic graph)
+adding edge and display functions in class along with Cycle to check for cycle
+and if no cycle exits then print a topological sort.
+*/
+class Graph {
+
+  int n = 0;
+  vector < node > * EdgeList;
+
+  public:
+    Graph() {
+      n = 0;
+    }
+
+  Graph(int s) { //intialise edgelist as vector of nodes
+    n = s;
+    EdgeList = new vector < node > [n];
+  }
+  void addEdge(int src, int dest) { //add edges
+
+    node newNode;
+    newNode.dest = dest;
+
+    EdgeList[src].push_back(newNode);
+
+  }
+
+  bool Cycle(int v, vector < int > & visited, vector < int > & rec) { //Check cycle
+
+    if (!visited[v]) {
+
+      visited[v] = 1;
+      rec[v] = 1;
+
+      for (auto it: EdgeList[v]) {
+
+        if (!visited[it.dest] && Cycle(it.dest, visited, rec)) return true;
+
+        else if (rec[it.dest]) return true;
+
+      }
+    }
+
+    rec[v] = 0;
+    return false;
+  }
+
+  void TopSort(int v, vector < int > & visited, stack < int > & s) { //Topological sorted elements added in stack
+
+    visited[v] = 1;
+
+    for (auto it: EdgeList[v]) {
+
+      if (!visited[it.dest]) TopSort(v, visited, s); //visit graph in order(depth first) for topological
+
+    }
+
+    s.push(v);
+
+  }
+
+  void display() { //display edges
+
+    for (int i = 0; i < n; i++) {
+      for (auto it: EdgeList[i]) {
+
+        cout << it.dest << "\n";
+      }
+    }
+
+  }
+};
+
+int main() {
+
+  int flag = 0, n, k, src, dest;
+  cout << "\nEnter no. of nodes\n"; //input number of nodes
+  cin >> n;
```

Comment:
your code terminates when I test it 
Check it where and fix it first

Reply:
Yes some changes needed,  looking into it

Reply:
Can you share the link to check for code formatting i.e tabs and spaces.Apart from changes like adding name and link to readme  and code specific changes anything else required?

Reply:
yes apart from readme correct the spelling of initialize in line no 22

Reply:
Code is not yet present on main so what link should I add